package kaggle;

message Competition {
  required string id = 1;
  optional string title = 2;
  optional string description = 6;
  optional string url = 3;

  optional Date start = 4;
  optional Date end = 5;

  enum Attribute {
    KNOWLEDGE = 1;
    LIMITED = 2;
  }

  repeated Attribute attributes = 7;
  optional int32 reward_usd = 8;
}

message Date {
  optional int64 timestamp_utc = 1;
}

message CalendarSyncerConfig {
  message ServiceAccount {
    optional string email = 1;
    optional string keysFile = 2;
  }

  message Authentication {
    optional string developersKey = 1;
    optional ServiceAccount serviceAccount = 2;
  }

  message Target {
    optional string email = 1;
    optional string calendarId = 2;
  }

  optional Authentication authentication = 1;
  optional Target target = 2;
}
